spring-integration-zeromq/src/main/java/org/springframework/integration/zeromq/inbound/ZeroMqMessageProducer.java

```diff
@@ -284,7 +300,14 @@ private Mono<Message<?>> convertMessage(Mono<ZMsg> msgMono) {
 		return msgMono.map((msg) -> {
 			Map<String, Object> headers = null;
 			if (msg.size() > 1) {
-				headers = Collections.singletonMap(ZeroMqHeaders.TOPIC, msg.unwrap().getString(ZMQ.CHARSET));
+				ZFrame topicFrame;
+				if (this.unwrapTopic) {
```

Comment:
Can we have this as a ternary operator instead of `if..else` ? 
I mean this is really just simple one like of logic.

Reply:
Of course, `if..else` was just for readability, but ternary operator seems much more expressive!